lib/rpmte.cc

```diff
@@ -190,16 +176,15 @@ static int addTE(rpmte p, Header h, fnpyKey key, rpmRelocation * relocs)
     p->pkgFileSize = 0;
     p->headerSize = headerSizeof(h, HEADER_MAGIC_NO);
 
-    p->thisds = rpmdsThisPool(tspool, h, RPMTAG_PROVIDENAME, RPMSENSE_EQUAL);
-    p->provides = rpmdsNewPool(tspool, h, RPMTAG_PROVIDENAME, 0);
-    p->requires_ = rpmdsNewPool(tspool, h, RPMTAG_REQUIRENAME, 0);
-    p->conflicts = rpmdsNewPool(tspool, h, RPMTAG_CONFLICTNAME, 0);
-    p->obsoletes = rpmdsNewPool(tspool, h, RPMTAG_OBSOLETENAME, 0);
-    p->order = rpmdsNewPool(tspool, h, RPMTAG_ORDERNAME, 0);
-    p->recommends = rpmdsNewPool(tspool, h, RPMTAG_RECOMMENDNAME, 0);
-    p->suggests = rpmdsNewPool(tspool, h, RPMTAG_SUGGESTNAME, 0);
-    p->supplements = rpmdsNewPool(tspool, h, RPMTAG_SUPPLEMENTNAME, 0);
-    p->enhances = rpmdsNewPool(tspool, h, RPMTAG_ENHANCENAME, 0);
+    p->dependencies[RPMTAG_NAME] = \
+	rpmdsThisPool(tspool, h, RPMTAG_PROVIDENAME, RPMSENSE_EQUAL);
+    for (rpmTagVal tag : {
+	    RPMTAG_PROVIDENAME, RPMTAG_SUPPLEMENTNAME, RPMTAG_ENHANCENAME,
+	    RPMTAG_REQUIRENAME, RPMTAG_RECOMMENDNAME, RPMTAG_SUGGESTNAME,
+	    RPMTAG_CONFLICTNAME, RPMTAG_OBSOLETENAME, RPMTAG_ORDERNAME
+	}) {
```

Comment:
Don't use code for what you can easily put into a data structure. Here's a nice opportunity to do just that.